lib/sparse/benches/search.rs

```diff
@@ -0,0 +1,191 @@
+use std::io;
+use std::path::Path;
+use std::sync::atomic::AtomicBool;
+
+use common::types::PointOffsetType;
+use criterion::{criterion_group, criterion_main, Criterion};
+use dataset::Dataset;
+use indicatif::{ProgressBar, ProgressDrawTarget};
+use itertools::Itertools;
+use rand::rngs::StdRng;
+use rand::SeedableRng as _;
+use sparse::common::scores_memory_pool::ScoresMemoryPool;
+use sparse::common::sparse_vector::SparseVector;
+use sparse::common::sparse_vector_fixture::{random_positive_sparse_vector, random_sparse_vector};
+use sparse::index::inverted_index::inverted_index_ram::InvertedIndexRam;
+use sparse::index::inverted_index::inverted_index_ram_builder::InvertedIndexBuilder;
+use sparse::index::loaders::{self, Csr};
+use sparse::index::search_context::SearchContext;
+mod prof;
+
+const NUM_QUERIES: usize = 2048;
+const MAX_SPARSE_DIM: usize = 30_000;
+const TOP: usize = 10;
+
+pub fn bench_search(c: &mut Criterion) {
+    bench_uniform_random(c, "random-50k", 50_000);
+    bench_uniform_random(c, "random-500k", 500_000);
+
+    let query_vectors =
+        loaders::load_csr_vecs(Dataset::AnnChallengeQueries.download().unwrap()).unwrap();
+
+    let index_1m = load_csr_index(Dataset::AnnChallenge1M.download().unwrap(), 1.0).unwrap();
```

Comment:
It's the same benchmark used in https://github.com/qdrant/sparse-vectors-benchmark. It's embeddings based on MS MARCO. Renamed to `NeurIps2023` as it would be accurate name.